themes-default/slim/src/store/modules/notifiers.js

```diff
@@ -0,0 +1,211 @@
+import { ADD_CONFIG } from '../mutation-types';
+
+const state = {
```

Comment:
You may need to play around with this but I think it'll be better off if we split the notifers into modules.

<details>
 <summary>Click me for diff</summary>

```diff
diff --git a/themes-default/slim/src/store/modules/notifiers.js b/themes-default/slim/src/store/modules/notifiers.js
deleted file mode 100644
index b92ce2a69..000000000
--- a/themes-default/slim/src/store/modules/notifiers.js
+++ /dev/null
@@ -1,211 +0,0 @@
-import { ADD_CONFIG } from '../mutation-types';
-
-const state = {
-    emby: {
-        enabled: null,
-        host: null,
-        apiKey: null
-    },
-    kodi: {
-        enabled: null,
-        alwaysOn: null,
-        libraryCleanPending: null,
-        cleanLibrary: null,
-        host: [],
-        username: null,
-        password: null,
-        notifyOnSnatch: null,
-        notifyOnDownload: null,
-        notifyOnSubtitleDownload: null,
-        update: {
-            library: null,
-            full: null,
-            onlyFirst: null
-        }
-    },
-    plex: {
-        client: {
-            host: [],
-            username: null,
-            enabled: null,
-            notifyOnSnatch: null,
-            notifyOnDownload: null,
-            notifyOnSubtitleDownload: null
-        },
-        server: {
-            updateLibrary: null,
-            host: [],
-            enabled: null,
-            https: null,
-            username: null,
-            password: null,
-            token: null
-        }
-    },
-    nmj: {
-        enabled: null,
-        host: null,
-        database: null,
-        mount: null
-    },
-    nmjv2: {
-        enabled: null,
-        host: null,
-        dbloc: null,
-        database: null
-    },
-    synologyIndex: {
-        enabled: null
-    },
-    synology: {
-        enabled: null,
-        notifyOnSnatch: null,
-        notifyOnDownload: null,
-        notifyOnSubtitleDownload: null
-    },
-    pyTivo: {
-        enabled: null,
-        host: null,
-        name: null,
-        shareName: null
-    },
-    growl: {
-        enabled: null,
-        host: null,
-        password: null,
-        notifyOnSnatch: null,
-        notifyOnDownload: null,
-        notifyOnSubtitleDownload: null
-    },
-    prowl: {
-        enabled: null,
-        api: null,
-        messageTitle: null,
-        piority: null,
-        notifyOnSnatch: null,
-        notifyOnDownload: null,
-        notifyOnSubtitleDownload: null
-    },
-    libnotify: {
-        enabled: null,
-        notifyOnSnatch: null,
-        notifyOnDownload: null,
-        notifyOnSubtitleDownload: null
-    },
-    pushover: {
-        enabled: null,
-        apiKey: null,
-        userKey: null,
-        device: [],
-        sound: null,
-        notifyOnSnatch: null,
-        notifyOnDownload: null,
-        notifyOnSubtitleDownload: null
-    },
-    boxcar2: {
-        enabled: null,
-        notifyOnSnatch: null,
-        notifyOnDownload: null,
-        notifyOnSubtitleDownload: null,
-        accessToken: null
-    },
-    pushalot: {
-        enabled: null,
-        notifyOnSnatch: null,
-        notifyOnDownload: null,
-        notifyOnSubtitleDownload: null,
-        authToken: null
-    },
-    pushbullet: {
-        enabled: null,
-        notifyOnSnatch: null,
-        notifyOnDownload: null,
-        notifyOnSubtitleDownload: null,
-        authToken: null,
-        device: null
-    },
-    freemobile: {
-        enabled: null,
-        notifyOnSnatch: null,
-        notifyOnDownload: null,
-        notifyOnSubtitleDownload: null,
-        api: null,
-        id: null
-    },
-    telegram: {
-        enabled: null,
-        notifyOnSnatch: null,
-        notifyOnDownload: null,
-        notifyOnSubtitleDownload: null,
-        api: null,
-        id: null
-    },
-    twitter: {
-        enabled: null,
-        notifyOnSnatch: null,
-        notifyOnDownload: null,
-        notifyOnSubtitleDownload: null,
-        dmto: null,
-        username: null,
-        password: null,
-        prefix: null,
-        directMessage: null
-    },
-    trakt: {
-        enabled: null,
-        pinUrl: null,
-        username: null,
-        accessToken: null,
-        timeout: null,
-        defaultIndexer: null,
-        sync: null,
-        syncRemove: null,
-        syncWatchlist: null,
-        methodAdd: null,
-        removeWatchlist: null,
-        removeSerieslist: null,
-        removeShowFromApplication: null,
-        startPaused: null,
-        blacklistName: null
-    },
-    email: {
-        enabled: null,
-        notifyOnSnatch: null,
-        notifyOnDownload: null,
-        notifyOnSubtitleDownload: null,
-        host: null,
-        port: null,
-        from: null,
-        tls: null,
-        username: null,
-        password: null,
-        addressList: null,
-        subject: null
-    },
-    slack: {
-        enabled: null,
-        notifyOnSnatch: null,
-        notifyOnDownload: null,
-        notifyOnSubtitleDownload: null,
-        webhook: null
-    }
-};
-
-const mutations = {
-    [ADD_CONFIG](state, { section, config }) {
-        if (section === 'notifiers') {
-            state = Object.assign(state, config);
-        }
-    }
-};
-
-const getters = {};
-
-const actions = {};
-
-export default {
-    state,
-    mutations,
-    getters,
-    actions
-};
diff --git a/themes-default/slim/src/store/modules/notifiers/boxcar2.js b/themes-default/slim/src/store/modules/notifiers/boxcar2.js
new file mode 100644
index 000000000..073e47489
--- /dev/null
+++ b/themes-default/slim/src/store/modules/notifiers/boxcar2.js
@@ -0,0 +1,20 @@
+export const state = {
+    enabled: null,
+    notifyOnSnatch: null,
+    notifyOnDownload: null,
+    notifyOnSubtitleDownload: null,
+    accessToken: null
+};
+
+export const mutations = {};
+
+export const getters = {};
+
+export const actions = {};
+
+export default {
+    state,
+    mutations,
+    getters,
+    actions
+};
diff --git a/themes-default/slim/src/store/modules/notifiers/email.js b/themes-default/slim/src/store/modules/notifiers/email.js
new file mode 100644
index 000000000..2a84a1460
--- /dev/null
+++ b/themes-default/slim/src/store/modules/notifiers/email.js
@@ -0,0 +1,27 @@
+export const state = {
+    enabled: null,
+    notifyOnSnatch: null,
+    notifyOnDownload: null,
+    notifyOnSubtitleDownload: null,
+    host: null,
+    port: null,
+    from: null,
+    tls: null,
+    username: null,
+    password: null,
+    addressList: null,
+    subject: null
+};
+
+export const mutations = {};
+
+export const getters = {};
+
+export const actions = {};
+
+export default {
+    state,
+    mutations,
+    getters,
+    actions
+};
diff --git a/themes-default/slim/src/store/modules/notifiers/emby.js b/themes-default/slim/src/store/modules/notifiers/emby.js
new file mode 100644
index 000000000..aa89d2392
--- /dev/null
+++ b/themes-default/slim/src/store/modules/notifiers/emby.js
@@ -0,0 +1,18 @@
+export const state = {
+    enabled: null,
+    host: null,
+    apiKey: null
+};
+
+export const mutations = {};
+
+export const getters = {};
+
+export const actions = {};
+
+export default {
+    state,
+    mutations,
+    getters,
+    actions
+};
diff --git a/themes-default/slim/src/store/modules/notifiers/freemobile.js b/themes-default/slim/src/store/modules/notifiers/freemobile.js
new file mode 100644
index 000000000..e613b9902
--- /dev/null
+++ b/themes-default/slim/src/store/modules/notifiers/freemobile.js
@@ -0,0 +1,21 @@
+export const state = {
+    enabled: null,
+    notifyOnSnatch: null,
+    notifyOnDownload: null,
+    notifyOnSubtitleDownload: null,
+    api: null,
+    id: null
+};
+
+export const mutations = {};
+
+export const getters = {};
+
+export const actions = {};
+
+export default {
+    state,
+    mutations,
+    getters,
+    actions
+};
diff --git a/themes-default/slim/src/store/modules/notifiers/growl.js b/themes-default/slim/src/store/modules/notifiers/growl.js
new file mode 100644
index 000000000..a6eeec630
--- /dev/null
+++ b/themes-default/slim/src/store/modules/notifiers/growl.js
@@ -0,0 +1,21 @@
+export const state = {
+    enabled: null,
+    host: null,
+    password: null,
+    notifyOnSnatch: null,
+    notifyOnDownload: null,
+    notifyOnSubtitleDownload: null
+};
+
+export const mutations = {};
+
+export const getters = {};
+
+export const actions = {};
+
+export default {
+    state,
+    mutations,
+    getters,
+    actions
+};
diff --git a/themes-default/slim/src/store/modules/notifiers/index.js b/themes-default/slim/src/store/modules/notifiers/index.js
new file mode 100644
index 000000000..55374206e
--- /dev/null
+++ b/themes-default/slim/src/store/modules/notifiers/index.js
@@ -0,0 +1,66 @@
+import { ADD_CONFIG } from '../../mutation-types';
+import boxcar2 from './boxcar2';
+import email from './email';
+import emby from './emby';
+import growl from './growl';
+import kodi from './kodi';
+import libnotify from './libnotify';
+import nmj from './nmj';
+import nmjv2 from './nmjv2';
+import plex from './plex';
+import prowl from './prowl';
+import pushalot from './pushalot';
+import pushbullet from './pushbullet';
+import pushover from './pushover';
+import pyTivo from './py-tivo';
+import slack from './slack';
+import synology from './synology';
+import synologyIndex from './synology-index';
+import telegram from './telegram';
+import trakt from './trakt';
+import twitter from './twitter';
+
+const state = {};
+
+const mutations = {
+    [ADD_CONFIG](state, { section, config }) {
+        if (section === 'notifiers') {
+            state = Object.assign(state, config);
+        }
+    }
+};
+
+const getters = {};
+
+const actions = {};
+
+const modules = {
+    boxcar2,
+    email,
+    emby,
+    growl,
+    kodi,
+    libnotify,
+    nmj,
+    nmjv2,
+    plex,
+    prowl,
+    pushalot,
+    pushbullet,
+    pushover,
+    pyTivo,
+    slack,
+    synology,
+    synologyIndex,
+    telegram,
+    trakt,
+    twitter
+};
+
+export default {
+    state,
+    mutations,
+    getters,
+    actions,
+    modules
+};
diff --git a/themes-default/slim/src/store/modules/notifiers/kodi.js b/themes-default/slim/src/store/modules/notifiers/kodi.js
new file mode 100644
index 000000000..2eb6b9362
--- /dev/null
+++ b/themes-default/slim/src/store/modules/notifiers/kodi.js
@@ -0,0 +1,30 @@
+export const state = {
+    enabled: null,
+    alwaysOn: null,
+    libraryCleanPending: null,
+    cleanLibrary: null,
+    host: [],
+    username: null,
+    password: null,
+    notifyOnSnatch: null,
+    notifyOnDownload: null,
+    notifyOnSubtitleDownload: null,
+    update: {
+        library: null,
+        full: null,
+        onlyFirst: null
+    }
+};
+
+export const mutations = {};
+
+export const getters = {};
+
+export const actions = {};
+
+export default {
+    state,
+    mutations,
+    getters,
+    actions
+};
diff --git a/themes-default/slim/src/store/modules/notifiers/libnotify.js b/themes-default/slim/src/store/modules/notifiers/libnotify.js
new file mode 100644
index 000000000..d109b6c02
--- /dev/null
+++ b/themes-default/slim/src/store/modules/notifiers/libnotify.js
@@ -0,0 +1,19 @@
+export const state = {
+    enabled: null,
+    notifyOnSnatch: null,
+    notifyOnDownload: null,
+    notifyOnSubtitleDownload: null
+};
+
+export const mutations = {};
+
+export const getters = {};
+
+export const actions = {};
+
+export default {
+    state,
+    mutations,
+    getters,
+    actions
+};
diff --git a/themes-default/slim/src/store/modules/notifiers/nmj.js b/themes-default/slim/src/store/modules/notifiers/nmj.js
new file mode 100644
index 000000000..ebd06a1ec
--- /dev/null
+++ b/themes-default/slim/src/store/modules/notifiers/nmj.js
@@ -0,0 +1,19 @@
+export const state = {
+    enabled: null,
+    host: null,
+    database: null,
+    mount: null
+};
+
+export const mutations = {};
+
+export const getters = {};
+
+export const actions = {};
+
+export default {
+    state,
+    mutations,
+    getters,
+    actions
+};
diff --git a/themes-default/slim/src/store/modules/notifiers/nmjv2.js b/themes-default/slim/src/store/modules/notifiers/nmjv2.js
new file mode 100644
index 000000000..0620a1e9f
--- /dev/null
+++ b/themes-default/slim/src/store/modules/notifiers/nmjv2.js
@@ -0,0 +1,19 @@
+export const state = {
+    enabled: null,
+    host: null,
+    dbloc: null,
+    database: null
+};
+
+export const mutations = {};
+
+export const getters = {};
+
+export const actions = {};
+
+export default {
+    state,
+    mutations,
+    getters,
+    actions
+};
diff --git a/themes-default/slim/src/store/modules/notifiers/plex.js b/themes-default/slim/src/store/modules/notifiers/plex.js
new file mode 100644
index 000000000..b47326e88
--- /dev/null
+++ b/themes-default/slim/src/store/modules/notifiers/plex.js
@@ -0,0 +1,32 @@
+export const state = {
+    client: {
+        host: [],
+        username: null,
+        enabled: null,
+        notifyOnSnatch: null,
+        notifyOnDownload: null,
+        notifyOnSubtitleDownload: null
+    },
+    server: {
+        updateLibrary: null,
+        host: [],
+        enabled: null,
+        https: null,
+        username: null,
+        password: null,
+        token: null
+    }
+};
+
+export const mutations = {};
+
+export const getters = {};
+
+export const actions = {};
+
+export default {
+    state,
+    mutations,
+    getters,
+    actions
+};
diff --git a/themes-default/slim/src/store/modules/notifiers/prowl.js b/themes-default/slim/src/store/modules/notifiers/prowl.js
new file mode 100644
index 000000000..94071cd70
--- /dev/null
+++ b/themes-default/slim/src/store/modules/notifiers/prowl.js
@@ -0,0 +1,22 @@
+export const state = {
+    enabled: null,
+    api: null,
+    messageTitle: null,
+    piority: null,
+    notifyOnSnatch: null,
+    notifyOnDownload: null,
+    notifyOnSubtitleDownload: null
+};
+
+export const mutations = {};
+
+export const getters = {};
+
+export const actions = {};
+
+export default {
+    state,
+    mutations,
+    getters,
+    actions
+};
diff --git a/themes-default/slim/src/store/modules/notifiers/pushalot.js b/themes-default/slim/src/store/modules/notifiers/pushalot.js
new file mode 100644
index 000000000..c1a1c9ad7
--- /dev/null
+++ b/themes-default/slim/src/store/modules/notifiers/pushalot.js
@@ -0,0 +1,20 @@
+export const state = {
+    enabled: null,
+    notifyOnSnatch: null,
+    notifyOnDownload: null,
+    notifyOnSubtitleDownload: null,
+    authToken: null
+};
+
+export const mutations = {};
+
+export const getters = {};
+
+export const actions = {};
+
+export default {
+    state,
+    mutations,
+    getters,
+    actions
+};
diff --git a/themes-default/slim/src/store/modules/notifiers/pushbullet.js b/themes-default/slim/src/store/modules/notifiers/pushbullet.js
new file mode 100644
index 000000000..384453c8e
--- /dev/null
+++ b/themes-default/slim/src/store/modules/notifiers/pushbullet.js
@@ -0,0 +1,21 @@
+export const state = {
+    enabled: null,
+    notifyOnSnatch: null,
+    notifyOnDownload: null,
+    notifyOnSubtitleDownload: null,
+    authToken: null,
+    device: null
+};
+
+export const mutations = {};
+
+export const getters = {};
+
+export const actions = {};
+
+export default {
+    state,
+    mutations,
+    getters,
+    actions
+};
diff --git a/themes-default/slim/src/store/modules/notifiers/pushover.js b/themes-default/slim/src/store/modules/notifiers/pushover.js
new file mode 100644
index 000000000..237ff4400
--- /dev/null
+++ b/themes-default/slim/src/store/modules/notifiers/pushover.js
@@ -0,0 +1,23 @@
+export const state = {
+    enabled: null,
+    apiKey: null,
+    userKey: null,
+    device: [],
+    sound: null,
+    notifyOnSnatch: null,
+    notifyOnDownload: null,
+    notifyOnSubtitleDownload: null
+};
+
+export const mutations = {};
+
+export const getters = {};
+
+export const actions = {};
+
+export default {
+    state,
+    mutations,
+    getters,
+    actions
+};
diff --git a/themes-default/slim/src/store/modules/notifiers/py-tivo.js b/themes-default/slim/src/store/modules/notifiers/py-tivo.js
new file mode 100644
index 000000000..1b4bffde7
--- /dev/null
+++ b/themes-default/slim/src/store/modules/notifiers/py-tivo.js
@@ -0,0 +1,19 @@
+export const state = {
+    enabled: null,
+    host: null,
+    name: null,
+    shareName: null
+};
+
+export const mutations = {};
+
+export const getters = {};
+
+export const actions = {};
+
+export default {
+    state,
+    mutations,
+    getters,
+    actions
+};
diff --git a/themes-default/slim/src/store/modules/notifiers/slack.js b/themes-default/slim/src/store/modules/notifiers/slack.js
new file mode 100644
index 000000000..03fd3884c
--- /dev/null
+++ b/themes-default/slim/src/store/modules/notifiers/slack.js
@@ -0,0 +1,20 @@
+export const state = {
+    enabled: null,
+    notifyOnSnatch: null,
+    notifyOnDownload: null,
+    notifyOnSubtitleDownload: null,
+    webhook: null
+};
+
+export const mutations = {};
+
+export const getters = {};
+
+export const actions = {};
+
+export default {
+    state,
+    mutations,
+    getters,
+    actions
+};
diff --git a/themes-default/slim/src/store/modules/notifiers/synology-index.js b/themes-default/slim/src/store/modules/notifiers/synology-index.js
new file mode 100644
index 000000000..af1ed8d25
--- /dev/null
+++ b/themes-default/slim/src/store/modules/notifiers/synology-index.js
@@ -0,0 +1,16 @@
+export const state = {
+    enabled: null
+};
+
+export const mutations = {};
+
+export const getters = {};
+
+export const actions = {};
+
+export default {
+    state,
+    mutations,
+    getters,
+    actions
+};
diff --git a/themes-default/slim/src/store/modules/notifiers/synology.js b/themes-default/slim/src/store/modules/notifiers/synology.js
new file mode 100644
index 000000000..d109b6c02
--- /dev/null
+++ b/themes-default/slim/src/store/modules/notifiers/synology.js
@@ -0,0 +1,19 @@
+export const state = {
+    enabled: null,
+    notifyOnSnatch: null,
+    notifyOnDownload: null,
+    notifyOnSubtitleDownload: null
+};
+
+export const mutations = {};
+
+export const getters = {};
+
+export const actions = {};
+
+export default {
+    state,
+    mutations,
+    getters,
+    actions
+};
diff --git a/themes-default/slim/src/store/modules/notifiers/telegram.js b/themes-default/slim/src/store/modules/notifiers/telegram.js
new file mode 100644
index 000000000..e613b9902
--- /dev/null
+++ b/themes-default/slim/src/store/modules/notifiers/telegram.js
@@ -0,0 +1,21 @@
+export const state = {
+    enabled: null,
+    notifyOnSnatch: null,
+    notifyOnDownload: null,
+    notifyOnSubtitleDownload: null,
+    api: null,
+    id: null
+};
+
+export const mutations = {};
+
+export const getters = {};
+
+export const actions = {};
+
+export default {
+    state,
+    mutations,
+    getters,
+    actions
+};
diff --git a/themes-default/slim/src/store/modules/notifiers/trakt.js b/themes-default/slim/src/store/modules/notifiers/trakt.js
new file mode 100644
index 000000000..9d6d863ff
--- /dev/null
+++ b/themes-default/slim/src/store/modules/notifiers/trakt.js
@@ -0,0 +1,30 @@
+export const state = {
+    enabled: null,
+    pinUrl: null,
+    username: null,
+    accessToken: null,
+    timeout: null,
+    defaultIndexer: null,
+    sync: null,
+    syncRemove: null,
+    syncWatchlist: null,
+    methodAdd: null,
+    removeWatchlist: null,
+    removeSerieslist: null,
+    removeShowFromApplication: null,
+    startPaused: null,
+    blacklistName: null
+};
+
+export const mutations = {};
+
+export const getters = {};
+
+export const actions = {};
+
+export default {
+    state,
+    mutations,
+    getters,
+    actions
+};
diff --git a/themes-default/slim/src/store/modules/notifiers/twitter.js b/themes-default/slim/src/store/modules/notifiers/twitter.js
new file mode 100644
index 000000000..e7fdddcbc
--- /dev/null
+++ b/themes-default/slim/src/store/modules/notifiers/twitter.js
@@ -0,0 +1,24 @@
+export const state = {
+   enabled: null,
+    notifyOnSnatch: null,
+    notifyOnDownload: null,
+    notifyOnSubtitleDownload: null,
+    dmto: null,
+    username: null,
+    password: null,
+    prefix: null,
+    directMessage: null
+};
+
+export const mutations = {};
+
+export const getters = {};
+
+export const actions = {};
+
+export default {
+    state,
+    mutations,
+    getters,
+    actions
+};
```

</details>

Reply:
can you add these changes to the branch please?

Reply:
Personally i'm okay with having all these notifiers in one file. It doesn't bother me having them in one file. I think it's organized enough. But maybe there are other more technical reasons to split them?